File: GuessingGame/app/src/main/java/com/lambdaschool/guessinggame/MainActivity.java
package com.lambdaschool.guessinggame;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    static final int MAX_NUMBER = 101;
    static final int CORRECT    = 0;
    static final int LOW        = 1;
    static final int HIGH       = 2;
    static final int RESET      = 3;

    EditText  editEntry;
    Button    buttonSubmit;
//    ImageView imageResult;
    TextView  textMessage, textPrevious;

    int       target;
    Random    generator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editEntry    = findViewById(R.id.edit_entry);
        buttonSubmit = findViewById(R.id.button_submit);
//        imageResult  = findViewById(R.id.image_result);
        textMessage  = findViewById(R.id.text_message);
        textPrevious = findViewById(R.id.text_previous);
        generator    = new Random();

        buttonSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (buttonSubmit.getText() == getString(R.string.submit_button_text)) {
                    int guess = Integer.parseInt(editEntry.getText().toString());
                    editEntry.setText("");
                    textPrevious.setText(Integer.toString(guess));
                    updateUI(checkGuess(guess));
                } else {
                    target = generator.nextInt(MAX_NUMBER);
//                    imageResult.setImageResource(R.drawable.baseline_remove_24px);
                    updateUI(RESET);
                }
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        target = generator.nextInt(MAX_NUMBER);
    }

    private int checkGuess(int guess) {
        if (guess == target) {
            return CORRECT;
        } else if ( guess < target ) {
            return LOW;
        } else {
            return HIGH;
        }
    }

    private void updateUI(int result) {
        switch (result) {
            case CORRECT:
                textMessage.setText("Congratulations!");
//                imageResult.setImageResource(R.drawable.baseline_star_24px);
                buttonSubmit.setText(R.string.reset_button_text);
                break;
            case LOW:
                textMessage.setText("Guess is Too Low");
//                imageResult.setImageResource(R.drawable.baseline_arrow_upward_24px);
                break;
            case HIGH:
                textMessage.setText("Guess is Too High");
//                imageResult.setImageResource(R.drawable.baseline_arrow_downward_24px);
                break;
            case RESET:
                buttonSubmit.setText(R.string.submit_button_text);
                textMessage.setText(R.string.prompt);
//                imageResult.setImageResource(R.drawable.baseline_arrow_downward_24px);
                break;
        }
    }
}

File: GuessingGame/app/src/main/java/com/lambdaschool/guessinggame/MainActivity.kt
package com.lambdaschool.guessinggame

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.random.Random

class MainActivity : AppCompatActivity() {

    private var target: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button_submit.setOnClickListener {
            if (button_submit.text === getString(R.string.submit_button_text)) {
                val guess = Integer.parseInt(edit_entry.text.toString())
                edit_entry.setText("")
                text_previous.text = Integer.toString(guess)
                updateUI(checkGuess(guess))
            } else {
                target = Random.nextInt(MAX_NUMBER)
                updateUI(RESET)
            }
        }
    }

    override fun onStart() {
        super.onStart()
        target = Random.nextInt(MAX_NUMBER)
    }

    private fun checkGuess(guess: Int): Int {
        return if (guess == target) {
            CORRECT
        } else if (guess < target) {
            LOW
        } else {
            HIGH
        }
    }

    private fun updateUI(result: Int) {
        when (result) {
            CORRECT -> {
                text_message.text = "Congratulations!"
                button_submit.setText(R.string.reset_button_text)
            }
            LOW -> text_message.text = "Guess is Too Low"
            HIGH -> text_message.text = "Guess is Too High"
            RESET -> {
                button_submit.setText(R.string.submit_button_text)
                text_message.setText(R.string.prompt)
            }
        }
    }

    companion object {

        internal val MAX_NUMBER = 101
        internal val CORRECT = 0
        internal val LOW = 1
        internal val HIGH = 2
        internal val RESET = 3
    }
}
